util/perl/OpenSSL/paramnames.pm

```diff
@@ -785,46 +755,60 @@ sub locate_long_endings {
     return 0, '';
 }
 
-sub produce_decoder {
-    my $func_name = shift;
-    my @keys = @_;
-    my %t = generate_trie(@keys);
-    my $s;
-
-    locate_long_endings(\%t);
-
-    open local *STDOUT, '>', \$s;
-    printf "int %s(const char *s)\n{\n", $func_name;
-    generate_code_from_trie(0, \%t);
-    print "    return -1;\n}\n";
-    return $s;
-}
-
-sub produce_param_list {
-    my $static_array = shift;
-    my $array_name = shift;
-    my $static_func = shift;
-    my $func_name = shift;
+sub output_param_decoder {
+    my $decoder_name_base = shift;
     my @params = @_;
     my @keys = ();
-    my $s;
+    my %prms = ();
 
-    open local *STDOUT, '>', \$s;
-    print $static_array . ' ' if $static_array ne '';
-    printf "const OSSL_PARAM %s[] = {\n", $array_name;
+    print "/* Machine generated by util/perl/OpenSSL/paramnames.pm */\n";
+    # Output ettable param array
+    printf "#ifndef %s_ettable\n", $decoder_name_base;
```

Comment:
What is the point of the ifndef? Would it be better to just not compile in this case?